pallets/delegation/src/lib.rs

```diff
@@ -650,32 +651,40 @@ pub mod pallet {
 			Ok(Some(<T as Config>::WeightInfo::remove_delegation(removal_checks)).into())
 		}
 
-		/// Transfer the storage deposit from one account to another.
+		/// Changes the deposit owner.
 		///
-		/// If the currently required deposit is different, the new deposit
-		/// value will be reserved.
+		/// The balance that is reserved by the current deposit owner will be
+		/// freed and balance of the new deposit owner will get reserved.
 		///
 		/// The subject of the call must be the owner of the delegation node.
 		/// The sender of the call will be the new deposit owner.
-		#[pallet::weight(<T as Config>::WeightInfo::transfer_deposit())]
-		pub fn transfer_deposit(origin: OriginFor<T>, delegation_id: DelegationNodeIdOf<T>) -> DispatchResult {
+		#[pallet::weight(<T as Config>::WeightInfo::change_deposit_owner())]
+		pub fn change_deposit_owner(origin: OriginFor<T>, delegation_id: DelegationNodeIdOf<T>) -> DispatchResult {
 			let source = <T as Config>::EnsureOrigin::ensure_origin(origin)?;
 
-			let mut delegation = DelegationNodes::<T>::get(&delegation_id).ok_or(Error::<T>::DelegationNotFound)?;
+			let delegation = DelegationNodes::<T>::get(&delegation_id).ok_or(Error::<T>::DelegationNotFound)?;
 
 			// Deposit can only be swapped by the owner of the delegation node, not the
 			// parent or another ancestor.
 			ensure!(delegation.details.owner == source.subject(), Error::<T>::AccessDenied);
```

Comment:
Not touched in this PR, but should we change this error to be consistent with the other pallets?

Reply:
yes will be done in a separate PR